src/linux/disk.rs

```diff
@@ -26,36 +28,56 @@ pub enum DiskType {
 impl From<isize> for DiskType {
     fn from(t: isize) -> DiskType {
         match t {
-            0 => DiskType::HDD,
-            1 => DiskType::SSD,
+            0 => DiskType::SSD,
+            1 => DiskType::HDD,
             id => DiskType::Unknown(id),
         }
     }
 }
 
-pub fn new_disk(name: &str, mount_point: &str, file_system: &str) -> Disk {
+fn find_type_for_name(name: &OsStr) -> DiskType
+{
+    /* turn "sda1" into "sda": */
```

Comment:
`sda1`, `sda2`, etc. refer to different partitions on the same physical disk; the property of rotational or not is on the disk, not on individual partitions, which are not listed in `/sys/block/`. I don't believe the code worked previously.